### PR TITLE
[Odysee] Create presence for odysee.com

### LINKED ISSUE #3625: Odysee | odysee.com
**Prerequisites and essential questions** 
- [ ] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [x] Is the site a paid service? (e.g. Netflix, Hulu)
_(creators can upload videos who require $LBC)_
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 
FeverZ#9668 _(if you have some questions about the site)_

**What's the name of the service?** 
https://odysee.com/ | Odysee

**What should the Presence display?** 
Odysee is a alternative to Youtube working with the blockchain. 
Display the current video watching with the creator name and button to go watch the video.

**If possible, please provide a logo for the service (512x512 minimum)** 
odysee

### PR DESCRIPTION
**Pull request by mlunax**
Resolve #3625

### CODE REVIEW

File: websites/O/Odysee/presence.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+          }
+        }
+      }
+    }
+    if (presenceData.details == null) {
+      presence.setTrayTitle();
+      presence.setActivity();
+    } else {
+      presence.setActivity(presenceData);
+    }
```
Comment (SlowLife1661) [inline websites/O/Odysee/presence.ts:116]: Never null

File: websites/O/Odysee/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  "service": "Odysee",
+  "altnames": ["LBRY"],
+  "description": {
+    "en": "Web frontend for LBRY. LBRY does to publishing, what Bitcoin did to money."
+  },
+  "url": "odysee.com",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/GEL8nqd.png",
+  "thumbnail": "https://odysee.com/public/img/odysee.png",
```
Comment (SlowLife1661) [inline websites/O/Odysee/dist/metadata.json:14]: Upload to imgur

File: websites/O/Odysee/presence.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
+const presence = new Presence({ clientId: "837270687638224906" }),
+  presenceData: PresenceData = {
+    details: "Odysee",
+    largeImageKey: "logo"
+  };
```
Comment (SlowLife1661) [inline websites/O/Odysee/presence.ts:5]: You can just put this inside so you don't have to bother `delete x`

File: websites/O/Odysee/presence.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      ".content__viewer--floating"
+    );
+    if (document.location.pathname === "/" && !floatingViewer) {
+      presenceData.details = "Browsing homepage";
+      delete presenceData.smallImageKey;
+      delete presenceData.smallImageText;
+      delete presenceData.buttons;
+      delete presenceData.endTimestamp;
+      delete presenceData.startTimestamp;
+      delete presenceData.state;
```
Comment (ririxidev) [inline websites/O/Odysee/presence.ts:19]: oh man what is this?

File: websites/O/Odysee/presence.ts [hunk 5]
```diff
@@ -0,0 +1,118 @@
+const presence = new Presence({ clientId: "837270687638224906" }),
+  presenceData: PresenceData = {
+    details: "Odysee",
+    largeImageKey: "logo"
+  };
+presence.on("UpdateData", async () => {
+  const buttons = await presence.getSetting("buttons");
+  if (document.location.hostname === "odysee.com") {
```
Comment (ririxidev) [inline websites/O/Odysee/presence.ts:8]: dont need this check

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/O/Odysee/presence.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+              url: uploaderUrlElement.href
+            }
+          ];
+        }
+        if (video) {
+          presenceData.endTimestamp = presence.getTimestampsfromMedia(video)[1];
+          presenceData.smallImageKey = video.paused ? "paused" : "play";
+          presenceData.smallImageText = video.paused ? "Paused" : "Watching";
+          if (video.paused) {
+            delete presenceData.startTimestamp;
```
Comment (ririxidev) [inline websites/O/Odysee/presence.ts:95]: You don't set `startTimestamp` anywhere so you don't need to delete it.

File: websites/O/Odysee/presence.ts [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        presenceData.state =
+          uploaderName.textContent +
+          (uploaderTag ? " " + uploaderTag.textContent : "");
+        presenceData.smallImageKey = "paused";
+        presenceData.smallImageText = "Paused";
+
+        if (uploaderUrlElement && buttons) {
+          presenceData.buttons = [
+            {
+              label: "Watch video",
```
Comment (ririxidev) [inline websites/O/Odysee/presence.ts:81]:
```suggestion
              label: "Watch Video",
```

File: websites/O/Odysee/presence.ts [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+              url: document.URL
+            },
+            {
+              label: "View author's channel",
```
Comment (ririxidev) [inline websites/O/Odysee/presence.ts:85]:
```suggestion
              label: "View Channel",
```

**Review by ririxidev — CHANGES_REQUESTED** [review]

**Review by ririxidev — APPROVED** [review]

File: websites/O/Odysee/dist/metadata.json [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  "service": "Odysee",
+  "altnames": ["LBRY"],
+  "description": {
+    "en": "Web frontend for LBRY. LBRY does to publishing, what Bitcoin did to money."
+  },
+  "url": "odysee.com",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/GEL8nqd.png",
+  "thumbnail": "https://i.imgur.com/NQjEv5C.png",
```
Comment (Bas950) [inline websites/O/Odysee/dist/metadata.json:14]: thumbnail should be a promotional banner or a screenshot of the website homepage.

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by ririxidev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**